tests/components/media_player/test_directv.py

```diff
@@ -0,0 +1,494 @@
+
+
+class mockDIRECTVClass:
+    """A fake DirecTV DVR device."""
+
+    def __init__(self, ip, port=8080, clientAddr='0'):
+        """Initialize the fake DirecTV device."""
+        self._host = ip
+        self._port = port
+        self._device = clientAddr
+        self._standby = True
+        self._play = False
+
+        _LOGGER.debug("INIT with host: %s, port: %s, device: %s",
+                      self._host, self._port, self._device)
+
+        self._locations = LOCATIONS
+
+        self.attributes = LIVE
+
+    def get_locations(self):
+        """Mock for get_locations method."""
+        _LOGGER.debug("get_locations called")
+        test_locations = {
+            'locations': self._locations,
+            'status': {
+                'code': 200,
+                'commandResult': 0,
+                'msg': 'OK.',
+                'query': '/info/getLocations'
+            }
+        }
+
+        return test_locations
+
+    def get_standby(self):
+        """Mock for get_standby method."""
+        _LOGGER.debug("STANDBY is: %s", self._standby)
+        return self._standby
+
+    def get_tuned(self):
+        """Mock for get_tuned method."""
+        _LOGGER.debug("get_tuned called")
+        if self._play:
+            self.attributes['offset'] = self.attributes['offset']+1
+
+        test_attributes = self.attributes
+        test_attributes['status'] = {
+            "code": 200,
+            "commandResult": 0,
+            "msg": "OK.",
+            "query": "/tv/getTuned"
+        }
+        return test_attributes
+
+    def key_press(self, keypress):
+        """Mock for key_press method."""
+        _LOGGER.debug("Key Press: %s", keypress)
+        if keypress == 'poweron':
+            self._standby = False
+            self._play = True
+        elif keypress == 'poweroff':
+            self._standby = True
+            self._play = False
+        elif keypress == 'play':
+            self._play = True
+        elif keypress == 'pause' or keypress == 'stop':
+            self._play = False
+
+    def tune_channel(self, source):
+        """Mock for tune_channel method."""
+        _LOGGER.debug("Change channel %s", source)
+        self.attributes['major'] = int(source)
+
+
+async def test_setup_platform_config(hass):
+    """Test setting up the platform from configuration."""
+    with MockDependency('DirectPy'), \
+            patch('DirectPy.DIRECTV', new=mockDIRECTVClass):
+
+        await async_setup_component(hass, mp.DOMAIN, WORKING_CONFIG)
+        await hass.async_block_till_done()
+
+    assert len(hass.data[DATA_DIRECTV]) == 1
+    assert (IP_ADDRESS, DEFAULT_DEVICE) in hass.data[DATA_DIRECTV]
+
+
+async def test_setup_platform_discover(hass):
+    """Test setting up the platform from discovery."""
+    add_entities = Mock()
+    with MockDependency('DirectPy'), \
+            patch('DirectPy.DIRECTV', new=mockDIRECTVClass):
+
+        setup_platform(hass, {}, add_entities,
```

Comment:
Set up the component but without the platform config and then fire the correct discovery event to test discovery.

Reply:
We might need to set up discovery component and not the media_player component.

Reply:
Nah, let's just call `async_load_platform` from helpers.discovery.py.

Reply:
An empty dict as fifth argument to `async_load_platform` should work as hass config in this case.

Reply:
Just seeing your message. Not passing an empty dict but instead just one with component in it.
Can change to empty if you want me to. Let me know. :-)